pkg/controller/route/route_controller.go

```diff
@@ -216,6 +216,7 @@ func (rc *RouteController) reconcile(nodes []*v1.Node, routes []*cloudprovider.R
 				// Delete the route.
 				go func(route *cloudprovider.Route, startTime time.Time) {
 					defer wg.Done()
+					rateLimiter <- struct{}{}
```

Comment:
Do we need a 
```go
<-rateLimiter
```
When DeleteRoute is done?

Reply:
Yes - it's need to be added similarly like we do in line 178

Reply:
I apologize @andrewsykim  i didn't realize your question was not fully answered.

Rate limits is asserted twice, before create and before delete. *all* create  and *delete* in reality will run concurrently. The modification I added asserts before *delete* just as it does for *add* because both calls translate to *write* (PUT || DELETE) in whoever implements the Routes() interface.

On a side note (for a later PR). There is a good chance the reconciliation loop fail if *add* conflicts with something that is about to get deleted. It will recover in next reconciliation loop. In other words, we should always delete then add.

Reply:
Maybe I'm missing something, if we don't `<- rateLimiter`, wouldn't this block forever after 200 delete requests? See maxConcurrentRouteCreations.

Reply:
wow! how did i miss that *scratches head*..

Thank you for this. I added the receive part as needed.